third_party/move/move-compiler-v2/src/lib.rs

```diff
@@ -232,6 +245,30 @@ fn add_default_optimization_pipeline(pipeline: &mut FunctionTargetPipeline) {
     pipeline.add_processor(Box::new(UnreachableCodeRemover {}));
 }
 
+/// Disassemble the given compiled units and return the disassembled code as a string.
+pub fn disassemble_compiled_units(units: &[CompiledUnit]) -> anyhow::Result<String> {
+    let disassembled_units: anyhow::Result<Vec<_>> = units
+        .iter()
+        .map(|unit| {
+            let view = match unit {
+                CompiledUnit::Module(module) => BinaryIndexedView::Module(&module.module),
+                CompiledUnit::Script(script) => BinaryIndexedView::Script(&script.script),
+            };
+            Disassembler::from_view(view, location::Loc::new(FileHash::empty(), 0, 0))
+                .and_then(|d| d.disassemble())
+        })
+        .collect();
```

Comment:
Why not do the `?` here?  The var type will be simplified.

Reply:
Not sure I understand what you are saying here? Use `?` where?

I am using the pattern from here: https://doc.rust-lang.org/rust-by-example/error/iter_result.html#fail-the-entire-operation-with-collect

Reply:
Yes, but you have a `?` on the following use of this value to return the error immediately. Why not here?

Reply:
In order to use the pattern I linked above. Maybe you can show me a compilable version of what you are suggesting?

Reply:
The change you suggest above is not compilable.

Reply:
hmm.  Well, never mind, then.